src/main/java/put/io/black/java/logic/ScenarioManager.java

```diff
@@ -0,0 +1,298 @@
+        int maxNesting = 1;
+        if (firstLevelNodes.size() != 0) {
+            for (Node firstLevelNode : firstLevelNodes) {
+                if (firstLevelNode.getChildrenCount() != 0) {
+                    int level = searchTheNestingNode(firstLevelNode, maxNesting);
+                    if (level > maxNesting) {
+                        maxNesting = level;
+                    }
+                }
+            }
+            return maxNesting;
+        } else {
+            return 0;
+        }
+    }
+
+    private int searchTheNestingNode(Node node, int maxNestingLevel) {
+        if (node.getChildrenCount() != 0) {
+            for (Node child : node.getChildren()) {
+                int level = searchTheNestingNode(child, maxNestingLevel);
+                if (level > maxNestingLevel) {
+                    maxNestingLevel = level;
+                }
+            }
+            return maxNestingLevel;
+        }
+        return node.getNestingLevel();
+    }
+
+    public int countNumberOfScenarioSteps() {
+        int scenarioSteps = 0;
+        if (firstLevelNodes.size() != 0) {
+            for (Node firstLevelNode : firstLevelNodes) {
+                scenarioSteps++;
+                if (firstLevelNode.getChildrenCount() != 0) {
+                    scenarioSteps = searchTheNode(firstLevelNode, scenarioSteps);
+                }
+            }
+            return scenarioSteps;
+        } else {
+            return 0;
+        }
+    }
+
+    private int searchTheNode(Node node, int scenarioSteps) {
+        if (node.getChildrenCount() != 0) {
+            for (Node child : node.getChildren()) {
+                scenarioSteps = searchTheNode(child, ++scenarioSteps);
+            }
+        }
+        return scenarioSteps;
+    }
+
+    public int countKeyWordsInScenario() {
+        int keyWordsInScenario = 0;
+        if (firstLevelNodes.size() != 0) {
+            for (Node firstLevelNode : firstLevelNodes) {
+                if (firstLevelNode.getChildrenCount() != 0) {
+                    keyWordsInScenario = searchTheKeyWordsNode(firstLevelNode, keyWordsInScenario);
+                }
+            }
+            return keyWordsInScenario;
+        } else {
+            return 0;
+        }
+    }
+
+    private int searchTheKeyWordsNode(Node node, int keyWordsInScenario) {
+        if (node.getChildrenCount() != 0) {
+            for (Node child : node.getChildren()) {
+                keyWordsInScenario = searchTheNode(child, keyWordsInScenario);
+            }
+        }
+        if (lineStartFromKeyWord(node.getLine())) {
+            keyWordsInScenario++;
+        }
+        return keyWordsInScenario;
+    }
+
+    public String cutActorsFromScenario() {
+        LinkedList<String> scenarioWithoutActors = new LinkedList<>();
+        scenarioWithoutActors.addLast("");
+        for (Node firstLevelNode : firstLevelNodes) {
+            if (firstLevelNode.getChildrenCount() != 0) {
```

Comment:
You check `getChildrenCount() != 0` in few places. Maybe prepare new method `isChildrenEmpty()` to check this? Easy use java `isEmpty()` inside. Or maybe better. Check not empty -> `hasChildren()` method.